tests/components/roborock/conftest.py

```diff
@@ -225,4 +226,8 @@ def cleanup_map_storage(
             pathlib.Path(hass.config.path(tmp_path)) / mock_roborock_entry.entry_id
         )
         yield storage_path
+        # We need to first unload the config entry because unloading it will
+        # persist any unsaved maps to storage.
+        if mock_roborock_entry.state is ConfigEntryState.LOADED:
+            await hass.config_entries.async_unload(mock_roborock_entry.entry_id)
```

Comment:
I thought that this already happened in the tests itself? cc @epenet

Reply:
Yes, but I guess @allenporter wants to make sure it gets unloaded BEFORE the call to `rmtree`.

Reply:
Yes, thats right.

I am also thinking as an alternative this could be improved to update all the tests to mock out storage, which is what I did with local calendar:

https://github.com/home-assistant/core/blob/84ae476b678fa0e593e83a01db16359ca021189b/tests/components/local_calendar/conftest.py#L35